logprep/framework/rule_tree/rule_tree.py

```diff
@@ -212,8 +213,9 @@ def get_matching_rules(
             if child.does_match(event):
                 current_node = child
 
-                if current_node.matching_rules:
-                    matches += child.matching_rules
+                for matching_rule in current_node.matching_rules:
+                    if matching_rule not in matches:
+                        matches.append(matching_rule)
```

Comment:
if `matches` would be declared as a `set` this wouldn't be needed.

Reply:
You said that already and I commented it. If there is something I'm missing then please tell me.

Reply:
sorry... didn't see your comment